### DISCUSSION TITLE
Possible way to generate OpenAPI specification on the fly.

### DISCUSSION THREAD

**Discussion by iamkarshe** [mikecao/flight#520] [Ideas]
Once all routes are made, based on method and route prefix, generating Open API specification and build the documentation in folder.

- All routes with payloads and return type.
- Authentication and tokens.
- Like how FastAPI does with its applications.

**Comment by n0nag0n** [comment]
Hmm........this could be cool. I would do it as a separate package in the flightphp org. You can just run the `Flight::getRoutes()` and it would fetch all the routes and then you could build the specification at least based on that....it would be lacking comments and examples and such.....have you implemented anything related to this on your own and how'd it go?
